### PR TITLE
belindas-closet-nestjs_4_66_add-unit-tests-for-archive-functionality

### LINKED ISSUE #66: Unit Tests for Product Controller and Services Modules: Add unit tests for archive functionality
As a developer, I would like to implement some unit tests so that we can automate them during CI/CD to enforce better code quality.

*Describe the solution you'd like*

- Add unit tests for archive functionality for the product controller and services modules

Relates to #45

### PR DESCRIPTION
**Pull request by heosman**
Resolves #66 

This PR adds unit tests for archive functionality for the product controller and services modules.

Test results for products.controller.spec.ts:
products controller spec ts results

Test results for products.service.spec.ts:
products service spec ts results

Relates to #45

### CODE REVIEW

**Comment by CourtneyHoppus** [comment]
@heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo

File: src/products/services/products/products.service.spec.ts [hunk 1]
```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
         .mockReturnValue({
           exec: jest.fn().mockResolvedValueOnce(deletedProduct)
         } as any);
-      const result = await service.delete(mockProduct.id);
-      expect(model.findByIdAndUpdate).toHaveBeenCalledWith(
-        mockProduct.id,
-        { isHidden: true }
-      );
-      expect(result).toEqual(deletedProduct);
+      mockProduct = await service.delete(mockProduct.id);
```
Comment (CourtneyHoppus) [inline src/products/services/products/products.service.spec.ts:158]: You want to leave this as before, `const result = await service.delete(mockProduct.id)` as the mockProduct is already defined and we want to see what the service brings back.

File: src/products/services/products/products.service.spec.ts [hunk 2]
```diff
@@ -154,12 +155,24 @@ describe('ProductsService', () => {
...
+      expect(mockProduct).toEqual(deletedProduct);
```
Comment (CourtneyHoppus) [inline src/products/services/products/products.service.spec.ts:159]: Here we still want to compare the `result` to the `deletedProduct`, so it should still be `expect(result).toEqual(deletedProduct);` to use the `result` from the service.

**Review by CourtneyHoppus — CHANGES_REQUESTED** [review]

**Comment by taylorpapke** [comment]
> @heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo

I would prefer if this was done by @Robel-003 @tinpham5614 or @CourtneyHoppus as they have permission to use our issue naming convention for this and can create the correct issue to target this in a separate PR.

**Comment by heosman** [comment]
> > @heosman it looks like when the delete functionality was changed in pull request #64 that the testing wasn't updated. What you have here doesn't test it fully. But I don't think this is your issue and should probably be addressed in it's own issue. @brinkbrink @taylorpapke @theGaryLarson @tinpham5614 @Robel-003 @Seiyaroo
> 
> I would prefer if this was done by @Robel-003 @tinpham5614 or @CourtneyHoppus as they have permission to use our issue naming convention for this and can create the correct issue to target this in a separate PR.

My mistake, I'll close this pull request then.

**Review by taylorpapke — APPROVED** [review]
All tests are passing for me just like in your screenshots. @CourtneyHoppus have all the changes that you requested been made?

**Review by CourtneyHoppus — CHANGES_REQUESTED** [review]
Still reviewing this, but I can see some of the tests aren't quite right and/or complete.
I will continue to work on this.

**Comment by heosman** [comment]
> @heosman please see the PR #71 I updated the testing for the delete function and all of the work you did on the archive function was based on that. One thing I changed was that when you check for the Bad Request Exception, you use an invalid id and the check in the function catches that automatically. Checking for the Not Found Exception should resolve to a null and we should still check that the model was called, as that is what the function is doing.

@CourtneyHoppus I looked over the changes in your pull request for the delete function test and implemented them in the archive function test. Let me know if it looks alright!

**Comment by CourtneyHoppus** [comment]
Redirected to #73

**Pull request closed without merging**